plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/engine/AIEngine.java

```diff
@@ -75,4 +77,9 @@ Flow.Publisher<AIEngineResponseChunk> requestCompletionStream(
     boolean hasValidConfiguration() throws DBException;
 
     boolean isLoggingEnabled() throws DBException;
+
+    boolean isSuportTranscription();
+
+    @NotNull
+    TranscriptResult createSpeechTranscription(@NotNull Path audioFile) throws DBException;
```

Comment:
This is a potentially time-consuming operation. Please supply a `DBRProgressMonitor` there.

Just a nitpick, but having the API accept a `Path` quite limits its potential usage. Can it instead accept an `InputStream`?

Reply:
Actually, the caller already wraps this method in a Job with a DBRProgressMonitor. Would that be enough?

Reply:
A monitor opens up many possibilities. You can report the upload process and the transcription process. So it's worth sticking one there regardless.